runbotics/runbotics-scheduler/src/websocket/bot/bot.gateway.ts

```diff
@@ -113,6 +113,10 @@ export class BotWebSocketGateway implements OnGatewayDisconnect, OnGatewayConnec
             this.logger.log('Restored user\'s executions-count because of process interruption');
         }
 
+        if (processInstance?.notificationUrl) {
+            await this.botProcessService.notifyAboutProcessInstanceStatus(processInstance);
+        }
```

Comment:
take this and email notification to a different function. Both share a common goal to notify about the result/error so combine them and make logic in this message controller more readable